lib/assert.js

```diff
@@ -931,6 +932,79 @@ assert.doesNotMatch = function doesNotMatch(string, regexp, message) {
   internalMatch(string, regexp, message, doesNotMatch);
 };
 
+const noop = () => {};
+
+const mustCallChecks = [];
+
+var attachExitHandler = 0;
+
+function runCallChecks(exitCode) {
+  const console = require('console');
+
+  const failed = mustCallChecks.filter((context) => {
+    context.messageSegment = `exactly ${context.exact}`;
+    return context.actual !== context.exact;
+  });
+
+  failed.forEach((context) => {
+    console.error('Mismatched %s function calls. Expected %s, actual %d.',
+                  context.name,
+                  context.messageSegment,
+                  context.actual);
+    console.error(context.stack.split('\n').slice(2).join('\n'));
+  });
+
+  if (exitCode !== 0) return;
+  if (failed.length) process.exitCode = 1;
+}
+
+assert.mustCall = function mustCall(fn, exact = 1) {
+  if (process._exiting)
+    throw new ERR_MUSTCALL_UNAVAILABLE_DURING_EXIT();
+  if (typeof fn === 'number') {
+    exact = fn;
+    fn = noop;
+  } else if (fn === undefined) {
+    fn = noop;
+  }
+
+  if (typeof exact !== 'number')
+    throw new ERR_INVALID_ARG_TYPE(`Invalid exact value: ${exact}`);
+
+  // eslint-disable-next-line no-restricted-syntax
+  const stackTrace = inspect(new Error());
+
+  const context = {
+    exact: exact,
+    actual: 0,
+    stack: stackTrace,
+    name: fn.name || '<anonymous>'
```

Comment:
I would use this function name instead of "anonymous":

```suggestion
    name: fn.name || 'mustCall'
```